lib.js

```diff
@@ -0,0 +1,76 @@
+let { readdir, stat } = require("node:fs/promises");
+let path = require("node:path");
+/** @import { FileFinderOptions } from "./types.ts" */
+
+exports.FileFinder = class FileFinder {
+	/**
+	 * @param {string} root
+	 * @param {FileFinderOptions} options?
+	 */
+	constructor(root, { skipDotfiles, filter }) {
+		this._root = root;
+
+		/**
+		 * @param {string} filename
```

Comment:
Minor observation: Elsewhere you had used `fileName`, which is a bit of an irksome inconsistency. I prefer `filename`.

Reply:
Resolved: ea1d2d38806ccccd30054281e10fe4a259deac0f